app/assets/javascripts/segment_history_graph.coffee

```diff
@@ -15,7 +15,7 @@ $ ->
   hidden_data = []
   first_non_null_attempt = Infinity
 
-  $.each gon.run.splits, (i, split) ->
+  $.each gon.run.raw_splits, (i, split) ->
```

Comment:
gon.run.raw_splits is supposed to have the complete splits with the history.  When I moved the full_splits loop in reset_graph.js it broke all of the splits due to trying to .filter a null.